stdpopsim/catalog/HomSap/__init__.py

```diff
@@ -1552,3 +1552,129 @@ def _AJ():
+        Gutenkunst et al.
+        A schematic of this model is outlined in Figure 3B of Gutenkunst et al.
+    """
+    populations = [
+        stdpopsim.Population(
+            id="YRI", description="Environmental Genomes YRI (West Africans)"),
+        stdpopsim.Population(
+            id="CEU", description="Environmental Genomes CEU (Europeans)"),
+        stdpopsim.Population(
+            id="CHB", description="Environmental Genomes CHB (East Asians)"),
+        stdpopsim.Population(
+            id="MXL", description="Environmental Genomes MXL (Mexican-Americans)"),
+    ]
+
+    citations = [stdpopsim.Citation(
+        author="Gutenkunst et al.",
+        year=2009,
+        doi="https://doi.org/10.1371/journal.pgen.1000695",
+        reasons={stdpopsim.CiteReason.DEM_MODEL})
+    ]
+
+    generation_time = 25
+
+    # First we set out the maximum likelihood values of the various parameters
+    # all Africa-related parameters are given in Table 1 and all other
+    # parameters are given in Table 2.
+    N_A = 7300
+    N_B = 2100
+    N_AF = 12300
+    N_EU0 = 1500
+    N_AS0 = 590
+    N_MX0 = 800
+    # Times are provided in years, so we convert into generations.
+    T_AF = 220e3 / generation_time
+    T_B = 140e3 / generation_time
+    T_EU_AS = 26.4e3 / generation_time
+    T_MX = 21.6e3 / generation_time
+    T_ADMIX = 0
+    # We need to work out the starting (diploid) population sizes based on
+    # the growth rates provided for these three populations.
+    r_EU = 0.0023
+    r_AS = 0.0037
+    r_MX = 0.005
+    N_EU = N_EU0 / math.exp(-r_EU * T_EU_AS)
+    N_AS = N_AS0 / math.exp(-r_AS * T_EU_AS)
+    N_MX = N_MX0 / math.exp(-r_MX * T_MX)
+    # Migration rates during the various epochs.
+    m_AF_B = 25e-5
+    m_AF_EU = 3e-5
+    m_AF_AS = 1.9e-5
+    m_EU_AS = 13.5e-5
+    # European admixture proportion in Mexican-Americans.
+    f_MX = 0.48
+    # Population IDs correspond to their indexes in the population
+    # configuration array. Therefore, we have 0=YRI, 1=CEU, 2=CHB, 3=MXL
+    # initially.
+    population_configurations=[
+        msprime.PopulationConfiguration(
+            initial_size=N_AF, metadata=populations[0].asdict()),
+        msprime.PopulationConfiguration(
+            initial_size=N_EU, growth_rate=r_EU, metadata=populations[1].asdict()),
+        msprime.PopulationConfiguration(
+            initial_size=N_AS, growth_rate=r_AS, metadata=populations[2].asdict()),
+        msprime.PopulationConfiguration(
+            initial_size=N_MX, growth_rate=r_MX, metadata=populations[3].asdict())
+        ]
+    migration_matrix=[
+        [      0, m_AF_EU, m_AF_AS, 0],
+        [m_AF_EU,       0, m_EU_AS, 0],
+        [m_AF_AS, m_EU_AS,       0, 0],
+        [      0,       0,       0, 0]
+        ]
+    demographic_events=[
+        # CEU and MXL admix with proportion f_MX at T_ADMIX
+        msprime.MassMigration(
+            time=T_ADMIX, source=3, destination=1, proportion=f_MX),
```

Comment:
Having `T_ADMIX = 0` here is going to produce undesireable behaviour with the SLiM engine, because it will be impossible to sample the MX population after the admixture event. Maybe we should bodge it so that T_ADMIX is one generation in the past? Thoughts @apragsdale, @jeromekelleher?

Reply:
In ms they noted that to model admixture in the MXL population they generated a 5th population that exists for 0 time.

Reply:
I agree having T_ADMIX exactly equal to zero is odd and will lead to weirdness. I wonder why have a MassMigration at time zero at all - wouldn't it be simpler to set up the initial conditions so that this MassMigration had already happened?

Reply:
Right, I was initially trying to explicitly follow what the authors did, but I see the hiccup now. If I were to preset the admixture event from CEU to MXL wouldn't I still need to specify a time at which this occurred?

Reply:
That is quite strange, and I didn't realize there was that quirk to this model. My first reaction is that we might not want to support this model in that case, because (1) it has this difficulty with parameterization that makes it difficult to faithfully implement, and (2) it's not very realistic, since admixture was more like ~10 generations ago. Probably doesn't matter much for the SFS, but it won't be correct for IBD or LD. People who run this model to test recent admixture methods could be really led astray by it. What do you all think?

Reply:
Thanks @apragsdale, those are very good points. We certainly don't want to include a model and then recommend against using it (for whatever reason). Maybe @RyanGutenkunst has an opinion here too?

Reply:
Aaron's correct: the model is this way because a difference of ~0.005 (in genetics) units of admixture timing doesn't matter to the SFS. And in dadi it was more computationally convenient to put it at 0. Since we have well-documented history here, we could easily push this event to 5-10 generations ago. If that's not kosher to modify the model, we can always leave it out. (It was inferred in a strange way anyways, integrating out the African pop.)

Reply:
@grahamgower it seems like it might be best to ditch this model because of all the quirks? I am happy to adjust the admixture time if you think it is worth keeping.

Reply:
How were you planning on using the model @David-Peede?

Reply:
I was personally going to use it so simulate Chr6 under different evolutionary histories in SLiM and see if I could recapitulate the patterns of variation we observe in the empirical data from the 1000 Genomes Project. But it seems like this might not be the best model for that (or be useful to others), perhaps AmericanAdmixture_4B11 might better fit my needs.

Reply:
Yes, that's likely a better bet.

Reply:
I am going to go ahead and close this PR if that's ok with you @grahamgower

Reply:
Yep, that's fine @David-Peede. It can always be reopened and picked up again in the future if we change our minds.